src/utils/computeUsersDistributions.ts

```diff
@@ -1,17 +1,68 @@
-import { ethers } from "ethers";
+import { ethers, providers } from "ethers";
 import { computeMerkleTree, fetchUsers, getAccumulatedEmission, userBalancesToUnclaimedTokens, sumRewards } from ".";
 import { commify, formatUnits } from "ethers/lib/utils";
 import { finishedEpochs } from "../ages/ages";
 import { SUBGRAPH_URL } from "../config";
 import { StorageService } from "./StorageService";
 import { getEpochFromNumber } from "./timestampToEpoch";
 import { epochNumberToAgeEpochString } from "../helpers";
+import { EpochConfig } from "../ages";
 
 export enum DataProvider {
   Subgraph = "subgraph",
   RPC = "rpc",
 }
 
+export const computeUsersDistributionsForEpoch = async (
+  epoch: EpochConfig,
+  provider: providers.BaseProvider,
+  storageService: StorageService,
+  force?: boolean
+) => {
+  console.log(`Compute users distribution for epoch ${epoch.epochNumber}`);
+
+  const usersBalances = await fetchUsers(SUBGRAPH_URL, epoch.finalBlock ?? undefined);
+  const usersAccumulatedRewards = (
+    await Promise.all(
+      usersBalances.map(async ({ address, balances }) => ({
+        address,
+        accumulatedRewards: sumRewards(
+          await userBalancesToUnclaimedTokens(balances, epoch.finalTimestamp, provider, storageService)
+        ).toString(),
+      }))
+    )
+  ).filter(({ accumulatedRewards }) => accumulatedRewards !== "0");
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { leaves, ...merkleTree } = computeMerkleTree(usersAccumulatedRewards);
+
+  const totalEmission = getAccumulatedEmission(epoch.epochNumber);
+
+  const { age: ageName, epoch: epochName } = epochNumberToAgeEpochString(epoch.epochNumber);
+
+  await storageService.writeUsersDistribution(
+    epoch.epochNumber,
+    {
+      age: ageName,
+      epoch: epochName,
```

Comment:
Why keeping this here?

Reply:
Just to keep stg human readable in the dumped json 😄